components/script/resources/media-controls.js

```diff
@@ -121,6 +283,10 @@
         this.elements[camelCase(id)] = this.controls.getElementById(id);
       });
 
+      // Replace standard range inputs with custom ones
+      this.customProgress = new CustomRangeInput(this.elements.progress);
+      this.customVolume = new CustomRangeInput(this.elements.volumeLevel);
```

Comment:
I would prefer to fix #22728 for all range inputs rather than create custom controls for media elements. @simonwuelker Is it realistic to adapt the code in this PR to fix #22728?

Reply:
Yes, I believe that should be possible. However, naively duplicating the `<script>` tag in each `<input type=range>` shadow tree will likely cause performance issues as seen in https://github.com/servo/servo/issues/37205.